### PR TITLE
Add dataclass models

### PR DESCRIPTION
**Pull request by jschlyter**
The current API only provides access to the GraphQL queries and fields. This PR adds code for a more pythonic data model based on dataclasses, resulting in an API less tightly coupled with the backend GraphQL representation. A client of `pypolestar` should not need to know anything about GraphQL, fields etc.

## Release Notes

- **New Features**
  - Introduced methods to retrieve car data based on VIN, including car information, battery status, and odometer readings.
  - Added a new target in the Makefile for running tests.

- **Enhancements**
  - New data classes for structured vehicle information, including car specifications, odometer data, and battery details.
  - Added utility functions for safely accessing and converting values from nested data structures.

- **Testing**
  - Implemented a comprehensive suite of unit tests for utility functions to ensure reliability and correctness.

- **Dependencies**
  - Updated requirements to include `pytest` for testing capabilities.

### CODE REVIEW

**Comment by jschlyter** [comment]
cc @wouterhardeman

File: custom_components/polestar_api/pypolestar/models.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    @classmethod
+    def from_dict(cls, data: GqlDict) -> Self:
+        return cls(
+            average_speed_km_per_hour=get_field_name_float(
+                "averageSpeedKmPerHour", data
+            ),
+            odometer_meters=get_field_name_int("odometerMeters", data),
+            trip_meter_automatic_km=get_field_name_float("tripMeterAutomaticKm", data),
+            trip_meter_manual_km=get_field_name_float("tripMeterManualKm", data),
+            event_update_timestamp=get_field_name_datetime(
```
[the remark this answers is not in the document]
Comment (jschlyter) [reply custom_components/polestar_api/pypolestar/models.py]: Yes, but of course - good catch. Fixed in 8b6449e98e4c3c533a2adf04e7fca674d6c235ef.

**Review by wouterhardeman — APPROVED** [review]

**Comment by jschlyter** [comment]
@coderabbitai review

File: custom_components/polestar_api/pypolestar/utils.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def get_field_name_date(field_name: str, data: GqlDict) -> date | None:
+    if (value := get_field_name_value(field_name, data)) and isinstance(value, str):
+        return date.fromisoformat(value)
+
+
+def get_field_name_datetime(field_name: str, data: GqlDict) -> datetime | None:
+    if (value := get_field_name_value(field_name, data)) and isinstance(value, str):
+        return datetime.fromisoformat(value)
```
[the remark this answers is not in the document]
Comment (jschlyter) [reply custom_components/polestar_api/pypolestar/utils.py:52]: I do want the caller to get the exception, but thanks for the suggestion.

**Pull request merged**